core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -699,21 +737,29 @@ public String lockAll(TableToken tableToken, String lockReason, boolean ignoreSn
             return REASON_SNAPSHOT_IN_PROGRESS;
         }
         // busy metadata is same as busy reader from user perspective
-        String lockedReason = REASON_BUSY_READER;
-        if (metadataPool.lock(tableToken)) {
-            lockedReason = writerPool.lock(tableToken, lockReason);
-            if (lockedReason == null) {
-                // not locked
-                if (readerPool.lock(tableToken)) {
-                    LOG.info().$("locked [table=`").utf8(tableToken.getDirName())
-                            .$("`, thread=").$(Thread.currentThread().getId())
-                            .I$();
-                    return null;
+        String lockedReason;
+
+        if (tableReaderMetadataPool.lock(tableToken)) {
+            if (sequencerMetadataPool.lock(tableToken)) {
```

Comment:
nit: locking sequencer metadata pool is not needed for non-Wal tables which is easy to read directly from the token.